src/main/java/eu/dissco/backend/domain/jsonapi/JsonApiData.java

```diff
@@ -10,15 +11,16 @@ public class JsonApiData {
   String type;
   JsonNode attributes;
 
-  public <T extends Record> JsonApiData(String id, String type, T domainObject, ObjectMapper mapper){
-    this.id = id;
-    this.type = type;
-    this.attributes = mapper.valueToTree(domainObject);
-  }
+  @JsonCreator
   public JsonApiData(String id, String type, JsonNode attributes){
```

Comment:
This is the source of the bug. I think because there are two constructors, the program doesn't know which one to use, and it throws an error when deserializing a request.

Reply:
ok clear